### PR TITLE
Features/commit events rest

### PR DESCRIPTION
**Pull request by woksin**
## Summary

First rough take on getting a RESTful API around the EventStore. Staying true to the gRPC protobuf contracts

### Added
- Rest endpoints around the event store in a controller
- /api/event/commit - Commits events
- /api/event/commitForAggregate - Commits aggregate events
- /api/event/fetchForAggregate - Fetch events for aggregate

### Changed
- Added a layer between the gRPC EventStore service and the Event Store which is a service that is a singleton per tenant. The new EventStore controller is also using it

### CODE REVIEW

**Comment by woksin** [comment]
I've tried to stay as true to the gRPC protobuf contracts we have already in terms of the requests and the responses. To be able to do that I had to make a couple of ugly proxies and have some not-so-nice conversion logic going on. They are at least something that can be improved upon in the future

I have not though about the security implications, auth or which port it should be listening to. We'd need to figure those things out

**Comment by woksin** [comment]
What should we do about the port and the security of these endpoints?

File: Source/Server/Startup.cs [hunk 1]
```diff
@@ -40,14 +45,23 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
             if (env.IsDevelopment())
             {
                 app.UseDeveloperExceptionPage();
+                app.UseSwagger();
```
Comment (einari) [inline Source/Server/Startup.cs:48]: I would suggest having swagger on always?  As you don't then need to set the runtime up for development and making the APIs more accessible to everyone.

**Review by einari — CHANGES_REQUESTED** [review]
Nice..  Excited to get WebAPIs here... 

A suggestion; rename the Controllers namespace to WebAPI. 
Further more, suggesting it could be its own project as well - then we'd structurally isolate it outside of the server startup. 

The reason I say so is that we could be adding more options like GraphQL as well and its nice to keep things separate.

File: Source/Server/Startup.cs (hunk 1 above)
Comment (woksin) [reply Source/Server/Startup.cs:48]: Sure, that we can do.

**Comment by einari** [comment]
> What should we do about the port and the security of these endpoints?

The port needs to be configurable, which I guess it is by default through the appSettings file. But we could source the configuration from environment variables - enabling us to have a port env variable. 

As for security - I suggest we come back to this question.
From our internal perspective of running it in the cluster - we don't need to expose the port right now.

**Comment by woksin** [comment]
@einari So is this okay to pull in now?

**Review by einari — APPROVED** [review]

**Pull request merged**